02-express-tutorial/app.js

```diff
@@ -1,26 +1,100 @@
 import express from "express";
 import data from "./data.js";
+import peopleRouter from "./public/routes-people.js";
+import cookieParser from "cookie-parser";
+
+const { products } = data;
 
 const app = express();
 const port = 3000;
 
+function logger(req, res, next) {
+  const now = new Date();
+  console.log(`Method: ${req.method} Url:${req.url}`);
+  next();
+}
+
 //middleware
 app.use(express.static("./public"));
+app.use(logger);
+app.use(express.urlencoded({ extended: false }));
+app.use(express.json());
+app.use(cookieParser());
+
+function auth(req, res, next) {
+  const cookiesName = req.cookies?.name;
+
+  if (cookiesName) {
+    req.user = cookiesName;
+    next();
+  } else {
+    res.status(404).json({ message: "Unauthorized" });
```

Comment:
The correct status code for unauthorized is 401.

```suggestion
    res.status(401).json({ message: "Unauthorized" });
```